src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs

```diff
@@ -202,6 +209,33 @@ static async ValueTask<QuicConnection> StartConnectAsync(QuicClientConnectionOpt
     /// </summary>
     public IPEndPoint LocalEndPoint => _localEndPoint;
 
+    /// <summary>
+    /// Returns the last known number of bidirectional stream that can be opened on this connection.
+    /// </summary>
+    public int AvailableBidirectionalStreamsCount => Volatile.Read(ref _availableBidirectionalStreamsCount);
+    /// <summary>
+    /// Returns the last known number of unidirectional stream that can be opened on this connection.
+    /// </summary>
+    public int AvailableUnidirectionalStreamsCount => Volatile.Read(ref _availableUnidirectionalStreamsCount);
+
+    /// <summary>
+    /// Occurres when an additional stream capacity has been released by the peer. Corresponds to receiving MAX_STREAMS frame.
+    /// </summary>
+    public event QuicConnectionStreamsAvailableEventHandler? StreamsAvailable;
+    private async void OnStreamsAvailable(int bidirectionalStreamsCountIncrement, int unidirectionalStreamsCountIncrement)
+    {
+        // Bail out early to avoid queueing work on the thread pool as well as event args instantiation.
+        if (StreamsAvailable is null)
+        {
+            return;
+        }
+
+        // Do not invoke user-defined event handler code on MsQuic thread.
+        await Task.CompletedTask.ConfigureAwait(ConfigureAwaitOptions.ForceYielding);
+
+        StreamsAvailable?.Invoke(this, new QuicConnectionStreamsAvailableEventArgs(bidirectionalStreamsCountIncrement, unidirectionalStreamsCountIncrement));
```

Comment:
We might want error handling here since this is calling user code on the thread pool

Reply:
Done